carto/auth.py

```diff
@@ -137,12 +136,15 @@ def send(self, relative_path, http_method, **requests_args):
         try:
             http_method, requests_args = self.prepare_send(http_method, **requests_args)
 
-            return super(APIKeyAuthClient, self).send(relative_path,
-                                                      http_method,
-                                                      **requests_args)
+            response = super(APIKeyAuthClient, self).send(relative_path, http_method, **requests_args)
         except Exception as e:
             raise CartoException(e)
 
+        if CartoRateLimitException.isResponseRateLimited(response):
```

Comment:
I guess we'll have to add this logic to the `NonVerifiedAPIKeyAuthClient` as well (it's used mainly in on-premises, for the cases when there's no SSL certificate installed in the server)